File: scripts/with_test_translation_outputs/Translation/Avatar/WizardLMTeam/WizardCoder-33B-V1.1/Python/32e2d04029db9d731e20e67653c98bbd2613e7b9/Experiment_Results/intermediate/SR/Translation/use_test/WizardCoder-33B-V1.1/Avatar/Python/Java/atcoder_ABC148_F.java
<｜begin▁of▁sentence｜>Below is an instruction that describes a task. Write a response that appropriately completes the request.

### Instruction:
Translate the following Python code to Java and enclose your solution inside ```java and ```.
A sample test case is provided below:

Test input:
5 4 5
1 2
1 3
1 4
1 5

Expected output:
1


```
from collections import Counter , defaultdict , deque
from heapq import heappop , heappush , heapify
import sys , bisect , math , itertools , pprint , fractions
sys.setrecursionlimit ( 10 ** 8 )
mod = 10 ** 9 + 7
INF = float ( 'inf' )
def inp ( ) : return int ( sys.stdin.readline ( ) )
def inpl ( ) : return list ( map ( int , sys.stdin.readline ( ).split ( ) ) )
n , ta , ao = inpl ( )
ta -= 1 ; ao -= 1
g = [ [ ] for i in range ( n ) ]
for i in range ( n - 1 ) :
    a , b = inpl ( )
    g [ a - 1 ].append ( b - 1 )
    g [ b - 1 ].append ( a - 1 )
ta_dist = [ None ] * n
ta_dist [ ta ] = 0
ao_dist = [ None ] * n
ao_dist [ ao ] = 0
def ta_dfs ( node ) :
    for v in g [ node ] :
        if ta_dist [ v ] != None : continue
        ta_dist [ v ] = ta_dist [ node ] + 1
        ta_dfs ( v )
def ao_dfs ( node ) :
    for v in g [ node ] :
        if ao_dist [ v ] != None : continue
        ao_dist [ v ] = ao_dist [ node ] + 1
        ao_dfs ( v )
ao_dfs ( ao )
ta_dfs ( ta )
res = 0
for i in range ( n ) :
    if ta_dist [ i ] > ao_dist [ i ] : continue
    res = max ( res , ao_dist [ i ] )
print ( res - 1 )


```

### Response:
Here is the Java version of the Python code:

```java
import java.util.*;

public class Main {
    static int n, ta, ao;
    static ArrayList<Integer>[] g;
    static int[] ta_dist, ao_dist;

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();
        ta = sc.nextInt() - 1;
        ao = sc.nextInt() - 1;
        g = new ArrayList[n];
        for (int i = 0; i < n; i++) {
            g[i] = new ArrayList<>();
        }
        for (int i = 0; i < n - 1; i++) {
            int a = sc.nextInt() - 1;
            int b = sc.nextInt() - 1;
            g[a].add(b);
            g[b].add(a);
        }
        ta_dist = new int[n];
        ao_dist = new int[n];
        Arrays.fill(ta_dist, -1);
        Arrays.fill(ao_dist, -1);
        ta_dist[ta] = 0;
        ao_dist[ao] = 0;
        ta_dfs(ta);
        ao_dfs(ao);
        int res = 0;
        for (int i = 0; i < n; i++) {
            if (ta_dist[i] > ao_dist[i]) continue;
            res = Math.max(res, ao_dist[i]);
        }
        System.out.println(res - 1);
    }

    static void ta_dfs(int node) {
        for (int v : g[node]) {
            if (ta_dist[v] != -1) continue;
            ta_dist[v] = ta_dist[node] + 1;
            ta_dfs(v);
        }
    }

    static void ao_dfs(int node) {
        for (int v : g[node]) {
            if (ao_dist[v] != -1) continue;
            ao_dist[v] = ao_dist[node] + 1;
            ao_dfs(v);
        }
    }
}
```

Please note that Java does not have built-in functions for some Python libraries such as `fractions`, `itertools`, `pprint`, `sys.setrecursionlimit`, etc. Also, Java does not have `None` keyword, instead it uses `null`. So, I replaced `None` with `-1`. <｜end▁of▁sentence｜>

File: dataset/Intermediate/Translation/Avatar/Python/code/atcoder_ABC148_F.py
import threading
import queue
from collections import Counter, defaultdict, deque
from heapq import heappop, heappush, heapify
import sys, bisect, math, itertools, pprint, fractions
sys.setrecursionlimit(10 ** 8)
mod = 10 ** 9 + 7
INF = float('inf')

def Func_inp_0():
    return int(sys.stdin.readline())

def inpl():
    return list(map(int, sys.stdin.readline().split()))
queue_inpl0 = queue.Queue()

def inpl_thread(queue):
    result = inpl()
    queue.put(result)
thread_inpl0 = threading.Thread(target=inpl_thread, args=(queue_inpl0,))
thread_inpl0.start()
thread_inpl0.join()
result_inpl0 = queue_inpl0.get()
(n, ta, ao) = result_inpl0
ta = ta - 1
ao = ao - 1
g = [[] for i in range(n)]

def loop_28_0(i, stop, step):
    if step == 0 or (step > 0 and i >= stop) or (step < 0 and i <= stop):
        return
    (a, b) = inpl()
    g[a - 1].append(b - 1)
    g[b - 1].append(a - 1)
    loop_28_0(i + step, stop, step)
loop_28_0(0, n - 1, 1)
ta_dist = [None] * n
ta_dist[ta] = 0
ao_dist = [None] * n
ao_dist[ao] = 0

def ta_dfs(node):
    for v in g[node]:
        if ta_dist[v] != None:
            continue
        ta_dist[v] = ta_dist[node] + 1
        ta_dfs(v)

def ao_dfs(node):
    for v in g[node]:
        if ao_dist[v] != None:
            continue
        ao_dist[v] = ao_dist[node] + 1
        ao_dfs(v)
ao_dfs(ao)
ta_dfs(ta)
res = 0
LoopChecker158 = 575
LoopChecker258 = 574
for LoopIndexOut in range(LoopChecker158 // LoopChecker258):
    for i in range(n):
        if ta_dist[i] > ao_dist[i]:
            continue
        res = max(res, ao_dist[i])
print(res - 1)